src/video/cocoa/SDL_cocoawindow.m

```diff
@@ -1738,13 +1738,19 @@ - (BOOL)acceptsFirstMouse:(NSEvent *)theEvent
         return SDL_SetError("%s", [[e reason] UTF8String]);
     }
 
+    [nswindow setColorSpace:[NSColorSpace sRGBColorSpace]];
```

Comment:
This caused a big performance drop on older macOS versions (we had it in there for like an hour but backed it out: https://github.com/libsdl-org/SDL/commit/fd85f5748db4852c3ee5f8fc8253c691c7ca89fc). I haven't observed a similar performance drop in macOS 12 with it, but at the very least I think this should be heavily tested and also gated behind a system version check.

Reply:
Hmm, okay. I'll try to figure out how to make an OS version check to make it only happen on macOS 12+, perhaps.

Reply:
I've only tested it recently on a discrete AMD GPU so I'm not confident yet that it's cost-free on all GPU setups.

Reply:
My tests have been on AMD GPUs and Apple M1/M2. Not sure there's anything else still supported by Apple to care about.

Reply:
Many Macs have integrated Intel GPUs.

Reply:
Is this maybe a bug in the Metal rendering, using sRGB when we should be using a different colorspace? I don't think we expect an sRGB colorspace in any of the SDL rendering code.

Reply:
This is mostly an OpenGL-specific change/fix actually, as far as I know: on Macs with P3 displays, the backbuffer colors of the opengl content view in the window are treated as 0-1 P3 instead of 0-1 (linear or encoded) sRGB, because the compositor just copies the backbuffer data to the display.

So without this change, OpenGL apps have oversaturated colors on P3 displays because the system is effectively reading P3 colorspace data from the window contents when the app didn't intend it.

Reply:
I'm not sure what colorspace OpenGL core is using on macOS internally but it's doing something close to Display P3, even though Apple asserts that their OpenGL implementation is colorspace agnostic. It ends up causing the colors to look a lot more saturated than they probably should (based on MoltenVK/Metal behavior, and comparison with another non-Mac system).